### ISSUE TITLE
pkg:analyzer uses a pkg:test import in the lib directory (without defining pkg:test as a regular dep)

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#35467]
```
[WARNING] build_modules|dartdevc on package:test_core/.dartdevc.meta_module.raw:
Unable to read module information for package:test, make sure you have a dependency on it in your pubspec.
```

Is this related to build or test? @grouma ?

**Comment by kevmoo** [comment]
@scheglov – couldn't `element_type_matchers` be moved to `test/shared_matchers.dart` or something?

**Comment by kevmoo** [comment]
@stereotype441 @scheglov  – could we bump the priority here? It's producing a lot of inexplicable `[WARNING]` lines for pretty common cases. Tough to explain to our users

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/88780

**Comment by nshahan** [comment]
I've been running into a wall of these warning messages for a few weeks now. Will an updated version be published soon?

**Comment by stereotype441** [comment]
@kevmoo @nshahan Sorry for the slow reply.  I'm preparing to publish an updated version of the analyzer now: https://dart-review.googlesource.com/c/sdk/+/89821

**Comment by stereotype441** [comment]
Ok, this is fixed in analyzer version 0.34.2.
